test/Test/Database/LSMTree/Normal/StateMachine.hs

```diff
@@ -1493,6 +1506,61 @@ updateStats action lookUp modelBefore _modelAfter result =
              }
         _ -> stats
 
+    updParentTable stats = case (action, result) of
+        (New{}, MEither (Right (MTableHandle tbl))) ->
+          stats {
+            parentTable = Map.insert (Model.tableHandleID tbl)
+                                     (Model.tableHandleID tbl)
+                                     (parentTable stats)
+          }
+        (Open{}, MEither (Right (MTableHandle tbl))) ->
+          stats {
+            parentTable = Map.insert (Model.tableHandleID tbl)
+                                     (Model.tableHandleID tbl)
+                                     (parentTable stats)
+          }
+        (Duplicate ptblVar, MEither (Right (MTableHandle tbl))) ->
+          let -- immediate and ultimate parent table ids
+              iptblId, uptblId :: Model.TableHandleID
+              iptblId = getTableHandleId (lookUp ptblVar)
+              uptblId = parentTable stats Map.! iptblId
+           in stats {
+                parentTable = Map.insert (Model.tableHandleID tbl)
+                                         uptblId
+                                         (parentTable stats)
+              }
+        _ -> stats
+
+    updDupTableActionLog stats = case action of
```

Comment:
Should this not also check whether the action was successful?

Reply:
Maybe, but we're hopefully going to be generating far fewer failing actions.